pkg/agent/manager/manager_test.go

```diff
@@ -1,160 +1,883 @@
+		return
+	}
+	if key != baseSVIDKey {
+		t.Error("PrivateKey is not equals to configured one")
+		return
+	}
+
+	// Loop until we detect a rotation
+	util.RunWithTimeout(t, 2*m.rotationFreq, func() {
+		for cert, _ = m.getBaseSVIDEntry(); cert.Equal(baseSVID); cert, _ = m.getBaseSVIDEntry() {
+		}
+	})
+
+	cert, key = m.getBaseSVIDEntry()
+	if cert.Equal(baseSVID) {
+		t.Error("SVID did not rotate")
+		return
+	}
+	if key == baseSVIDKey {
+		t.Error("PrivateKey did not rotate")
+		return
+	}
+}
+
+func TestSynchronization(t *testing.T) {
+	dir := createTempDir(t)
+	defer removeTempDir(dir)
+
+	trustDomain := "example.org"
+
+	apiHandler := newMockNodeAPIHandler(&mockNodeAPIHandlerConfig{
+		t:                 t,
+		trustDomain:       trustDomain,
+		dir:               dir,
+		fetchSVIDResponse: fetchSVIDResponseForTestSynchronization,
+		svidTTL:           2,
+	})
+	apiHandler.start()
+	defer apiHandler.stop()
+
+	baseSVID, baseSVIDKey := apiHandler.newSVID("spiffe://"+trustDomain+"/spire/agent/join_token/abcd", 1*time.Hour)
+
+	c := &Config{
+		ServerAddr: &net.UnixAddr{
+			Net:  "unix",
+			Name: apiHandler.sockPath,
+		},
+		SVID:          baseSVID,
+		SVIDKey:       baseSVIDKey,
+		Log:           testLogger,
+		TrustDomain:   url.URL{Host: trustDomain},
+		SVIDCachePath: path.Join(dir, "svid.der"),
+		Bundle:        apiHandler.bundle,
+		Tel:           &telemetry.Blackhole{},
+	}
+
+	mgr, err := New(c)
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	m := mgr.(*manager)
+	m.rotationFreq = 1 * time.Hour
+	m.syncFreq = 2 * time.Second
+	err = m.Start()
+	if err != nil {
+		t.Error(err)
+		return
+	}
+	defer m.Shutdown()
+
+	done := make(chan struct{})
+	wu := m.Subscribe(cache.Selectors{
+		&common.Selector{Type: "unix", Value: "uid:1111"},
+		&common.Selector{Type: "spiffe_id", Value: "spiffe://example.org/spire/agent/join_token/abcd"},
+	}, done)
+
+	// Before synchronization
+	entriesBefore := cacheEntriesAsMap(m.cache.Entries())
+	if len(entriesBefore) != 3 {
+		t.Error("3 cached entries were expected")
+		return
+	}
+
+	util.RunWithTimeout(t, 1*time.Second, func() {
+		u := <-wu
+
+		if len(u.Entries) != 3 {
+			t.Errorf("expected 3 entries, got: %d", len(u.Entries))
+		}
+
+		if len(u.Bundle) != 1 {
+			t.Error("expected 1 bundle")
+		}
+
+		if !u.Bundle[0].Equal(apiHandler.bundle[0]) {
+			t.Error("received bundle should be equals to the server bundle")
+		}
+
+		entriesUpdated := cacheEntriesAsMap(u.Entries)
+		for key, eu := range entriesUpdated {
+			eb, ok := entriesBefore[key]
+			if !ok {
+				t.Errorf("an update was received for an inexistent entry on the cache with EntryId=%v", key)
+				return
+			}
+			if eb != eu {
+				t.Error("entry received does not match entry on cache")
+				return
+			}
+		}
+	})
+
+	util.RunWithTimeout(t, 2*m.syncFreq, func() {
+		// There should be 3 updates after sync, because we are subcribed to selectors that
+		// matches with 3 entries that were renewed on the cache.
```

Comment:
Oh, I thought this was the expected behavior. Definitively we should add an issue to track this describing the expected behavior, I didn't add one because I'm not sure of what the expected behavior is.